src/Spago/Watch.hs

```diff
@@ -80,7 +80,7 @@ fileWatchConf watchConfig shouldClear inner = withManagerConf watchConfig $ \man
             -- and the last event either has different path, or has happened
             -- more than `debounceTime` seconds ago.
             let shouldRebuild =
-                   any matches (Set.toList globs)
+                   any matches globs
```

Comment:
Unrelated minor fix: I realized that `Set` of globs is also foldable, so `any :: Foldable t => (a -> Bool) -> t a -> Bool` will work with it and we can avoid this conversion to List for every filesystem change event.